modules/imap/hm-imap-bodystructure.php

```diff
@@ -274,6 +274,15 @@ private function parse_multi_part_subs($vals) {
             }
             else {
                 $subs = array_merge($subs, $this->id_parts(array($val)));
+                foreach ($subs as &$sub) {
+                    if (!$sub['size'] && is_numeric($sub['encoding'])) {
```

Comment:
id_parts seem to return an array with numeric indexes. I think you should change this code in id_single_part where single_format list of fields point to the wrong positions in $vals array. Please add the exception there either by reordering the indexes in single_format before using them or reordering the vals array. Please also add a comment describing why are we doing this change - it seems like an edge case or improperly formatted message.

Reply:
@kroky  changes made